praetorian_cli/handlers/chariot.py

```
@@ -1,6 +1,46 @@
+                choice = int(raw)
+            except ValueError:
+                click.echo(f'Error: invalid input: {raw}')
+                raise SystemExit(1)
+            if choice < 1 or choice > len(account_names):
+                click.echo(f'Error: must be between 1 and {len(account_names)}')
+                raise SystemExit(1)
+            sdk.accounts.assume_role(account_names[choice - 1])
+        else:
+            keychain.account = saved_account
+    except (KeyboardInterrupt, click.Abort, SystemExit):
+        raise SystemExit(0)
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)